src/components/ArtistList/ArtistList.js

```diff
@@ -1,70 +1,56 @@
-import React, { Fragment, useEffect, useState } from "react";
+import React, { Fragment, useEffect} from "react";
+import { useSelector,useDispatch } from "react-redux";
+
 import Select from "react-dropdown-select";
 import ReactPaginate from "react-paginate";
 
+import { comboActions } from "../../store/combo-slice";
+import {fetchData} from '../../store/data-slice'
+
 import ArtistCard from "../ArtistCard/ArtistCard";
 import "./ArtistList.css";
 
 const ArtistList = () => {
-  //defining states
-  const [artists, setArtists] = useState([]);
-  const [pageCount, setPageCount] = useState(0);
-  const [itemsPerPage,setItemsPerPage]=useState(10)
-
-  const options=[{value:5,label:'five items per page'},{value:10,label:'ten items per page'},{value:15,label:'fifteen items per page'},{value:20,label:'twenty items per page'}]
-
-  //change items per page on select
-const itemsPerPageHandler=(selectedOption)=>{
-  setItemsPerPage(selectedOption[0].value)
-}
-
-
-  //for the first time that the page loads so we have something to show
-  //calculates the total number of pages we need for pagination
+  const dispatch=useDispatch()
+
+ 
+  const itemsPerPage=useSelector((state )=>state.combo.itemsPerPage);
+  const artists=useSelector(state=>state.data.data)
+  const pageCount=useSelector(state=>state.data.pageCount)
+  const currentPage=useSelector(state=>state.combo.currentPage)
+  const options=useSelector(state=>state.combo.options)
+  
+  
   useEffect(() => {
+   
     const getArtistRequest = async () => {
-      const url = `http://localhost:8080/api/crews?page=1&size=${itemsPerPage}`;
-      
-      const response = await fetch(url);
-      const data = await response.json();
-      const total = data.page.totalElements;
-      setPageCount(Math.floor(total / itemsPerPage));
-      setArtists(data.content);
+      const url = `http://localhost:8080/api/crews?page=${currentPage}&size=${itemsPerPage}`;
```

Comment:
Isn't it better to extract this constant configs to a separate config file?

Reply:
it is not possible in this case.